### PR TITLE
Prepare to move OrchestrationTemplate to each provider

### PR DESCRIPTION
**Pull request by bzwei**
Move the template classes to the corresponding provider repository.
This is a step before the migration.

https://bugzilla.redhat.com/show_bug.cgi?id=1417021

This is a refactor work but a step planned to fix above bugzilla issue.

### CODE REVIEW

File: app/models/manageiq/providers/openstack/cloud_manager/orchestration_template.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
+class ManageIQ::Providers::Openstack::CloudManager::OrchestrationTemplate < ::OrchestrationTemplate
```
Comment (bzwei) [inline app/models/manageiq/providers/openstack/cloud_manager/orchestration_template.rb]: This is a class moved from core repository https://github.com/ManageIQ/manageiq/blob/master/app/models/orchestration_template_hot.rb#L1

The old file will be removed after the migration is done.

File: app/models/manageiq/providers/openstack/cloud_manager/vnfd_template.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+class ManageIQ::Providers::Openstack::CloudManager::VnfdTemplate < ::OrchestrationTemplate
```
Comment (bzwei) [inline app/models/manageiq/providers/openstack/cloud_manager/vnfd_template.rb]: Move from https://github.com/ManageIQ/manageiq/blob/master/app/models/orchestration_template_vnfd.rb#L1
The original file will be removed after migration is done.

@Ladas Should this class under `CloudManager` or `InfraManager`?

Comment (Ladas) [inline app/models/manageiq/providers/openstack/cloud_manager/vnfd_template.rb]: it should be under a cloud manager

**Comment by bzwei** [comment]
Will not fix the codeclimate issue here, since the task is simply to relocate the file.

**Comment by bzwei** [comment]
@miq-bot add_label refactoring
@miq-bot assign @Ladas

**Comment by Ladas** [comment]
ping @aufi can we merge this? The orchestration provisioning is in a half broken state, so we need to move forward with this. :-)

**Pull request merged**